### PR TITLE
BUILD: use rpath for locating dynamic libraries

### PR DESCRIPTION
**Pull request by chipitsine**
@davidebeatrici , can you have a look please ?

this "works for me", however, I'm not sure I'm following best practices. I'm bit lost in various rpath/runpath options of CMake

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,15 @@ set(CMAKE_C_STANDARD 99)
 set(TOP_DIRECTORY ${CMAKE_SOURCE_DIR})
 set(BUILD_DIRECTORY ${CMAKE_BINARY_DIR})
 
+#
+# use rpath for locating installed libraries
+#
+set(CMAKE_SKIP_BUILD_RPATH  FALSE)
```
Comment (davidebeatrici) [inline CMakeLists.txt:29]:
```suggestion
set(CMAKE_SKIP_BUILD_RPATH FALSE)
```

Comment (davidebeatrici) [reply CMakeLists.txt:29]: Please note that the option's value is `FALSE` by default: https://cmake.org/cmake/help/latest/variable/CMAKE_SKIP_BUILD_RPATH.html

File: CMakeLists.txt [hunk 2]
```diff
@@ -23,6 +23,15 @@ set(CMAKE_C_STANDARD 99)
...
+set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
```
Comment (davidebeatrici) [inline CMakeLists.txt:30]: Please note that the option's value is `FALSE` by default: https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_WITH_INSTALL_RPATH.html

File: CMakeLists.txt [hunk 3]
```diff
@@ -23,6 +23,15 @@ set(CMAKE_C_STANDARD 99)
...
+set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
```
Comment (davidebeatrici) [inline CMakeLists.txt:31]:
```suggestion
set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_LIBDIR})
```

**Comment by davidebeatrici** [comment]
Please squash the commits.

**Comment by chipitsine** [comment]
I partially applied.

this suggestion

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index cb325ada..cc04db30 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ set(BUILD_DIRECTORY ${CMAKE_BINARY_DIR})
 #
 set(CMAKE_SKIP_BUILD_RPATH FALSE)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_LIBDIR})
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)

```


did not work. rpath was empty.
maybe CMAKE_INSTALL_LIBDIR is defined later ?


other suggestions are good. done.

**Comment by davidebeatrici** [comment]
`GNUInstallDirs` has to be included.

**Comment by chipitsine** [comment]
btw, we include GNUInstallDirs on line 47.

but if I put directives under it (and use `set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_LIBDIR})`), I see runpath is set to relative "lib"


`Set runtime path of "/usr/local/libexec/softether/vpnserver/vpnserver" to "lib"`

and ... nothing works


```
ilia@ubuntu20-softether:~/softethervpn$ ldd /usr/local/libexec/softether/vpnserver/vpnserver
	linux-vdso.so.1 (0x00007ffee3ac0000)
	libcedar.so => not found
	libmayaqua.so => not found
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fed84d77000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fed84f78000)
```

```
ilia@ubuntu20-softether:~/softethervpn$ readelf -d /usr/local/libexec/softether/vpnserver/vpnserver | grep RUNP
 0x000000000000001d (RUNPATH)            Library runpath: [lib]
ilia@ubuntu20-softether:~/softethervpn$ 
```

**Comment by chipitsine** [comment]
I'm moving things under "UNIX" section

**Pull request merged**